ethapi/api.go

```diff
@@ -2329,7 +2329,8 @@ func (api *PublicDebugAPI) traceBlock(ctx context.Context, block *evmcore.EvmBlo
 		}
 		res, err := api.traceTx(ctx, tx, msg, txctx, block.Header(), statedb, config, nil)
 		if err != nil {
-			return nil, err
+			results[i] = &txTraceResult{TxHash: tx.Hash(), Error: err.Error()}
+			continue
```

Comment:
Before continue, `statedb.EndTransaction()` should be called as it is in the end of this for loop. Also limits for response size are not takiing efect because of this.